File: PF2-web1/web1-backend/src/main/java/com/example/demo/bean/ProblemDomain.java
package com.example.demo.bean;

import java.util.ArrayList;

public class ProblemDomain extends PdNode{
	private int problemdomain_no;	//领域编号
	private String problemdomain_name;	//领域名称
	private String problemdomain_shortname;	//名称缩写
	private String problemdomain_type;	//领域类型
	private String problemdomain_property;	//物理特性
	private int problemdomain_x;	//位置信息
	private int problemdomain_y;	
	private int problemdomain_h;	
	private int problemdomain_w;
	private String state;
	private ArrayList<Phenomenon> phes;
	public ProblemDomain() {
		super();
	}

	public ProblemDomain(ProblemDomain item) {
		problemdomain_no = item.problemdomain_no;
		problemdomain_name = item.problemdomain_name;
		problemdomain_shortname = item.problemdomain_shortname;
		problemdomain_type = item.problemdomain_type;
		problemdomain_property = item.problemdomain_property;
		problemdomain_x = item.problemdomain_x;
		problemdomain_y = item.problemdomain_y;
		problemdomain_h = item.problemdomain_h;
		problemdomain_w = item.problemdomain_w;
		state = item.state;
	}
	public void setPhes(ArrayList<Phenomenon> phes){
		this.phes = phes;
	}
	public ArrayList<Phenomenon> getPhes(){
		return this.phes;
	}
	public int getProblemdomain_no() {
		return problemdomain_no;
	}
	public void setProblemdomain_no(int problemdomain_no) {
		this.problemdomain_no = problemdomain_no;
	}
	public String getProblemdomain_name() {
		return problemdomain_name;
	}
	public void setProblemdomain_name(String problemdomain_name) {
		this.problemdomain_name = problemdomain_name;
	}
	public String getProblemdomain_shortname() {
		return problemdomain_shortname;
	}
	public void setProblemdomain_shortname(String problemdomain_shortname) {
		this.problemdomain_shortname = problemdomain_shortname;
	}
	public String getProblemdomain_type() {
		return problemdomain_type;
	}
	public void setProblemdomain_type(String problemdomain_type) {
		this.problemdomain_type = problemdomain_type;
	}
	public String getProblemdomain_property() {
		if(problemdomain_property==null) {
			problemdomain_property="GivenDomain";
		}
		return problemdomain_property;
	}
	public void setProblemdomain_property(String problemdomain_property) {
		this.problemdomain_property = problemdomain_property;
	}
	public int getProblemdomain_x() {
		return problemdomain_x;
	}
	public void setProblemdomain_x(int problemdomain_x) {
		this.problemdomain_x = problemdomain_x;
	}
	public int getProblemdomain_y() {
		return problemdomain_y;
	}
	public void setProblemdomain_y(int problemdomain_y) {
		this.problemdomain_y = problemdomain_y;
	}
	public int getProblemdomain_h() {
		return problemdomain_h;
	}
	public void setProblemdomain_h(int problemdomain_h) {
		this.problemdomain_h = problemdomain_h;
	}
	public int getProblemdomain_w() {
		return problemdomain_w;
	}
	public void setProblemdomain_w(int problemdomain_w) {
		this.problemdomain_w = problemdomain_w;
	}
	public String getState() {
		return state;
	}
	public void setState(String state) {
		this.state = state;
	}
	public int getNo() {
		return problemdomain_no;
	}

	public void setNo(int no) {
		this.problemdomain_no = no;
	}

	public String getName() {
		return problemdomain_name;
	}

	public void setName(String name) {
		this.problemdomain_name = name;
	}

	public String getShortname() {
		return problemdomain_shortname;
	}

	public void setShortname(String shortname) {
		this.problemdomain_shortname = shortname;
	}

	public String getType() {
		return problemdomain_type;
	}

	public void setType(String type) {
		this.problemdomain_type = type;
	}

	public String getProperty() {
		return problemdomain_property;
	}

	public void setProperty(String property) {
		this.problemdomain_property = property;
	}

	public int getX() {
		return problemdomain_x;
	}

	public void setX(int x) {
		this.problemdomain_x = x;
	}

	public int getY() {
		return problemdomain_y;
	}

	public void setY(int y) {
		this.problemdomain_y = y;
	}

	public int getH() {
		return problemdomain_h;
	}

	public void setH(int h) {
		this.problemdomain_h = h;
	}

	public int getW() {
		return problemdomain_w;
	}

	public void setW(int w) {
		this.problemdomain_w = w;
	}
}

File: PF2-web2/web2-vue/src/types/ProblemDomain.ts
import { Phenomenon } from "./Phenomenon";
import { Node } from "./Node";

export class ProblemDomain extends Node {
  problemdomain_no: number;
  problemdomain_name: string;
  problemdomain_shortname: string;
  problemdomain_type: string;
  problemdomain_property: string;
  problemdomain_x: number;
  problemdomain_y: number;
  problemdomain_h: number;
  problemdomain_w: number;
  phes: Phenomenon[];

  static newProblemDomain(no: number, name: string, shortName: string, type: string, property: string, x: number, y: number, w: number, h: number) {
    let pd = new ProblemDomain();
    pd.problemdomain_no = no;
    pd.problemdomain_name = name;
    pd.problemdomain_shortname = shortName;
    pd.problemdomain_type = type;
    pd.problemdomain_property = property;
    pd.problemdomain_x = x;
    pd.problemdomain_y = y;
    pd.problemdomain_h = h;
    pd.problemdomain_w = w;
    return pd;
  }

  static newProblemDomainWithOld(old: ProblemDomain, name: string, shortName: string, type: string, property: string) {
    let pd = new ProblemDomain();
    pd.problemdomain_no = old.problemdomain_no;
    pd.problemdomain_x = old.problemdomain_x;
    pd.problemdomain_y = old.problemdomain_y;
    pd.problemdomain_h = old.problemdomain_h;
    pd.problemdomain_w = old.problemdomain_w;

    pd.problemdomain_name = name;
    pd.problemdomain_shortname = shortName;
    pd.problemdomain_type = type;
    pd.problemdomain_property = property;
    return pd;
  }

  static copyProblemDomain(old: ProblemDomain) {
    let pd = new ProblemDomain();
    pd.problemdomain_no = old.problemdomain_no;
    pd.problemdomain_x = old.problemdomain_x;
    pd.problemdomain_y = old.problemdomain_y;
    pd.problemdomain_h = old.problemdomain_h;
    pd.problemdomain_w = old.problemdomain_w;

    pd.problemdomain_name = old.problemdomain_name;
    pd.problemdomain_shortname = old.problemdomain_shortname;
    pd.problemdomain_type = old.problemdomain_type;
    pd.problemdomain_property = old.problemdomain_property;
    return pd;
  }

  getNo() { return this.problemdomain_no; }
  getName() { return this.problemdomain_name; }
  getShortName() { return this.problemdomain_shortname; }
  getX() { return this.problemdomain_x; }
  getY() { return this.problemdomain_y; }
  getH() { return this.problemdomain_h; }
  getW() { return this.problemdomain_w; }
  getProperty() { return this.problemdomain_property; }
  getType() { return this.problemdomain_type; }

  setNo(no: number) { this.problemdomain_no = no; }
  setName(name: string) { this.problemdomain_name = name; }
  setShortName(shortName: string) { this.problemdomain_shortname = shortName; }
  setX(x: number) { this.problemdomain_x = x; }
  setY(y: number) { this.problemdomain_y = y; }
  setH(h: number) { this.problemdomain_h = h; }
  setW(w: number) { this.problemdomain_w = w; }
}
